app/data_providers/slate_providers/for_you_slate_provider.py

```diff
@@ -34,17 +34,24 @@ async def rank_corpus_items(
             self,
             items: List[CorpusItemModel],
             preferred_topics: List[TopicModel] = None,
-            recommendation_count: int = None,
+            user_impression_capped_list: List[CorpusItemModel] = None,
             *args,
-            **kwargs
+            **kwargs,
     ) -> List[CorpusItemModel]:
         """
         :param items: Candidate corpus items
         :param preferred_topics: Topics explicitly preferred by the user.
-        :param recommendation_count: Maximum recommendations to return.
+        :param user_impression_capped_list: List of impression capped CorpusItem ids for the user.
         :return: Ranks items based on topic preferences, with topic spreading.
         """
-        return rank_by_preferred_topics(items, preferred_topics, recommendation_count)
+        # These parameters have default value 'None' to match the signature of the base class, but they are required.
+        assert preferred_topics is not None
+        assert user_impression_capped_list is not None
+
+        items = rank_by_impression_caps(items, user_impression_capped_list)
+        items = spread_topics(items)
+        items = rank_by_preferred_topics(items, preferred_topics)
```

Comment:
@lfishhead This order is different than we had during code pairing. Does this look right to you?

I'll dump a spreadsheet to let Carolyn QA the content.